packages/components/src/components/Calculator/index.tsx

```diff
@@ -170,13 +180,37 @@ type Props = {
   valueWithContext: SqCalculatorValueWithContext;
 };
 
+export const CalculatorSampleCountValidation: React.FC<{
+  calculator: SqCalculatorValueWithContext;
+  children: React.ReactNode;
+}> = ({ calculator, children }) => {
+  const { sampleCount } = calculator.value;
+
+  const sampleCountIsInvalid =
+    sampleCount &&
+    (sampleCount < SAMPLE_COUNT_MIN || sampleCount > SAMPLE_COUNT_MAX);
+
+  return sampleCountIsInvalid ? (
+    <ErrorAlert
+      heading={`Calculator sampleCount must be between ${SAMPLE_COUNT_MIN} and ${SAMPLE_COUNT_MAX}. It is set to ${sampleCount}.`}
```

Comment:
I wonder if we could use error boundary for this. Then we won't have to inject a custom error component here.

But it's not very important.